internal/provider/service_account_tokens_resource.go

```diff
@@ -0,0 +1,222 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	unleash "github.com/Unleash/unleash-server-api-go/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ resource.Resource                = &serviceAccountTokensResource{}
+	_ resource.ResourceWithConfigure   = &serviceAccountTokensResource{}
+	_ resource.ResourceWithImportState = &serviceAccountTokensResource{}
+)
+
+func NewServiceAccountTokensResource() resource.Resource {
+	return &serviceAccountTokensResource{}
+}
+
+type serviceAccountTokensResource struct {
+	client *unleash.APIClient
+}
+
+type serviceAccountTokensResourceModel struct {
+	Id               types.Int64  `tfsdk:"id"`
+	ServiceAccountId types.Int64  `tfsdk:"service_account_id"`
+	Secret           types.String `tfsdk:"secret"`
+	Description      types.String `tfsdk:"description"`
+	ExpiresAt        types.String `tfsdk:"expires_at"`
+}
+
+func (r *serviceAccountTokensResource) Configure(ctx context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*unleash.APIClient)
+	if !ok {
+		return
+	}
+	r.client = client
+}
+
+func (r *serviceAccountTokensResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account_token"
+}
+
+func (r *serviceAccountTokensResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages service account tokens.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Description: "The ID of the service account.",
+				Computed:    true,
+			},
+			"service_account_id": schema.Int64Attribute{
+				Description: "The ID of the service account token.",
+				Required:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
```

Comment:
TIL "to mark the accessible properties as requiring a plan change. This means that any update to a service account token will trigger a delete and then recreate it from scratch"

Nice! I think this might be handy for https://github.com/Unleash/terraform-provider-unleash/issues/163 as well.

Reply:
Me too! Thank you ChatGPT *cough*

Oh cheers, might be valuable to be able to address that issue actually!